src/app/core/filter/filter.service.ts

```diff
@@ -0,0 +1,107 @@
+import { Injectable } from "@angular/core";
+import { EntitySchemaField } from "../entity/schema/entity-schema-field";
+import { ConfigService } from "../config/config.service";
+import { DataFilter } from "../entity-components/entity-subrecord/entity-subrecord/entity-subrecord-config";
+import { Entity } from "../entity/model/entity";
+import {
+  createFactory,
+  allParsingInstructions,
+  allInterpreters,
+  Filter,
+  compare,
+} from "@ucast/mongo2js";
+import moment from "moment";
+
+/**
+ * This server handles the filter objects.
+ */
+@Injectable({
+  providedIn: "root",
+})
+export class FilterService {
+  private filterFactory = createFactory(
+    allParsingInstructions,
+    allInterpreters,
+    { compare: this.extendedCompare.bind(this) }
+  ) as Filter;
+
+  constructor(private configService: ConfigService) {}
+
+  /**
+   * Builds a predicate for a given filter object.
+   * This predicate can be used to filter arrays of objects.
+   * ```javascript
+   * const predicate = this.filterService.getFilterPredicate(filterObj);
+   * const filtered = this.data.filter(predicate)
```

Comment:
If we are always using it like this, we could also pack it into the function here. If we sometimes use the predicate separately, this implementation looks good.

`const filteredData = this.filterService.applyFilter(data, filterObj);`

Reply:
I think it is cheaper, to not re-create filters every time and I also just started using the predicate on a single entity 😅 So I suggest to keep it like this for now.